firedrake/dmplex.pyx

```diff
@@ -2371,3 +2371,39 @@ def clear_adjacency_callback(PETSc.DM dm not None):
         dm.removeLabel("ghost_region")
         CHKERR(DMLabelDestroy(&label))
     CHKERR(DMPlexSetAdjacencyUser(dm.dm, NULL, NULL))
+
+
+def set_patch_residual(patch, function, ctx, is_snes=False, interior_facets=False):
```

Comment:
Should this really live in dmplex.pyx? Alternatively, should dmplex.pyx be renamed?